File: app/src/main/java/com/uramonk/androidtemplateapp/repository/IWeatherRepository.java
package com.uramonk.androidtemplateapp.repository;

import com.uramonk.androidtemplateapp.entity.WeatherEntity;

import rx.Observable;

/**
 * Created by uramonk on 2016/06/30.
 */
public interface IWeatherRepository {
    Observable<WeatherEntity> getWeather();
}

File: app/src/main/java/com/uramonk/androidtemplateapp/repository/IWeatherRepository.kt
package com.uramonk.androidtemplateapp.repository

import com.uramonk.androidtemplateapp.entity.WeatherEntity

import rx.Observable

/**
 * Created by uramonk on 2016/06/30.
 */
interface IWeatherRepository {
    fun getWeather(): Observable<WeatherEntity>
}
